Feat: Complete High-Frequency P2P Feature with Particle Sync and Dedicated README (EOSU-654)

JessTello opened the pull request:
This PR completes the implementation of the High-Frequency Peer-to-Peer feature that was partially developed. It also introduces a dedicated README for the High-Frequency sample, which previously reused the Peer-to-Peer README despite having different functionalities.

**Changes**
- Finalized High-Frequency P2P logic and UI adjustments
- This update introduces real-time particle effect synchronization between connected users in the High-Frequency Peer-to-Peer sample. When a user clicks on the screen, the click position is transmitted to the other connected instance, which then renders a particle effect at the same location. This feature demonstrates the high-frequency data exchange capabilities of the EOS Plugin for Unity.
- Added High-Frequency README with usage instructions
- Updated related prefabs, scripts, and sample scene configurations

JessTello committed:
Feat: Complete High-Frequency P2P Feature and Add Dedicated README (JIRA-ID)

- Finished implementation of High-Frequency Peer-to-Peer functionality enables real-time particle effect synchronization between connected users based on click positions
- Added dedicated README for High-Frequency sample (previously reused P2P README)
- Updated prefabs, scripts, and scene configurations

matt-clarke commented on Assets/Scripts/EOSHighFrequencyP2P.cs:
```diff
@@ -299,10 +299,40 @@ public ProductUserId HandleReceivedMessages()
                 if (!peerId.IsValid())
                 {
-                    Debug.LogErrorFormat("EOS P2PNAT HandleReceivedMessages: ProductUserId peerId is not valid!");
+                    Debug.LogError($"{nameof(EOSHighFrequencyPeer2PeerManager)} {nameof(HandleReceivedMessages)}: ProductUserId for '{0}' is not valid!");
```
```suggestion
                    Debug.LogError($"{nameof(EOSHighFrequencyPeer2PeerManager)} {nameof(HandleReceivedMessages)}: ProductUserId with value '{peerId}' is not valid!");
```

JessTello replied:
Fixed in the latest commit.

matt-clarke commented on Assets/Scripts/EOSHighFrequencyP2P.cs:
```diff
@@ -299,10 +299,40 @@ public ProductUserId HandleReceivedMessages()
+                    {
+                        Debug.LogWarning($"{nameof(EOSHighFrequencyPeer2PeerManager)} {nameof(HandleReceivedMessages)}: Malformed coordinate message received.");
+                    }
+                }
```
Not your change but can you delete all the commented out code below?

JessTello replied:
Deleted in the latest commit.

matt-clarke commented on Assets/Scripts/StandardSamples/UI/Peer2Peer/HighFrequency/UIHighFrequencyPeer2PeerMenu.cs:
```diff
@@ -44,21 +44,23 @@ public class UIHighFrequencyPeer2PeerMenu : SampleMenuWithFriends
         private ProductUserId currentChatProductUserId;
 
+        private Camera uiCamera;
+        private const string CoordinateMessagePrefix = "m";
```
Please make the other const public and use it in this file too, having one place to control it is best

JessTello replied:
Fixed in the latest commit.

matt-clarke commented on Assets/Scripts/StandardSamples/UI/Peer2Peer/HighFrequency/UIHighFrequencyPeer2PeerMenu.cs:
```diff
@@ -89,15 +91,21 @@ protected override void Update()
+                ParticlesOnClick();
+            }
+
+            if (Peer2PeerManager.sendActive && currentChatProductUserId != null)
```
The code above calls `HasValidCurrentProductId`, which already checks this
```suggestion
            if (Peer2PeerManager.sendActive)
```

JessTello replied:
Fixed in the latest commit.

matt-clarke commented on Assets/Scripts/StandardSamples/UI/Peer2Peer/HighFrequency/UIHighFrequencyPeer2PeerMenu.cs:
```diff
@@ -323,22 +322,48 @@ protected override void HideInternal()
+            if (!HasValidCurrentProductId())
+            {
+                return;
+            }
```
Please move this to the top of the function, it makes the return more visible and avoids unnecessary operations

JessTello replied:
Fixed in the latest commit.

matt-clarke commented on Assets/Scripts/StandardSamples/UI/Peer2Peer/HighFrequency/UIHighFrequencyPeer2PeerMenu.cs:
```diff
@@ -323,22 +322,48 @@ protected override void HideInternal()
+            else
+            {
+                return true;
+            }
```
```suggestion

            return true;
```

JessTello replied:
Fixed in the latest commit.

matt-clarke commented on Assets/Scripts/StandardSamples/UI/Peer2Peer/HighFrequency/UIHighFrequencyPeer2PeerMenu.cs:
```diff
@@ -323,22 +322,48 @@ protected override void HideInternal()
+        }
+        private void ForceInitialHFPacket()
+        {
+            if (HasValidCurrentProductId())
```
```suggestion
            if (!HasValidCurrentProductId())
```

JessTello replied:
Fixed in the latest commit.

matt-clarke commented on Assets/Scripts/StandardSamples/UI/Peer2Peer/HighFrequency/UIHighFrequencyPeer2PeerMenu.cs:
```diff
@@ -323,22 +322,48 @@ protected override void HideInternal()
             if (currentChatProductUserId == null || !currentChatProductUserId.IsValid())
             {
-                Debug.LogError("UIPeer2PeerMenu (SendOnClick): ProductUserId for '{0}' is not valid!");
+                Debug.LogError($"{nameof(UIHighFrequencyPeer2PeerMenu)} {nameof(HasValidCurrentProductId)}: ProductUserId for '{0}' is not valid!");
```
```suggestion
                Debug.LogError($"{nameof(UIHighFrequencyPeer2PeerMenu)} {nameof(HasValidCurrentProductId)}: ProductUserId with value '{currentChatProductUserId}' is not valid!");
```

JessTello replied:
Fixed in the latest commit.

matt-clarke commented on Assets/Readme/HighFrequencyP2PReadMe.txt:
Can you fix the spelling mistake in the file name?

JessTello replied:
Fixed in the latest commit.

JessTello committed:
Refactor: Fix typos, improve logs, and correct conditional logic

- Fixed spelling error in README filename
- Changed private variable to public for centralized control
- Removed obsolete commented code
- Improved logs to return exact values
- Corrected particles-on-click flow to avoid unnecessary operations
- Added missing negation in conditional statement

billfr-ApexSystems reviewed (approved):
Approved with changes requested by Matt
